src/in_memory_maintenance.rs

```diff
@@ -0,0 +1,25 @@
+use crate::dune_data_loading::load_dune_data_into_memory;
+use crate::models::in_memory_database::InMemoryDatabase;
+use std::sync::Arc;
+use std::time::Duration;
+
+const MAINTENANCE_INTERVAL: Duration = Duration::from_secs(3);
+
+pub async fn in_memory_database_maintaince(
+    memory_database: Arc<InMemoryDatabase>,
+    dune_download_file: String,
+) {
+    let db = Arc::clone(&memory_database);
+    loop {
+        {
+            let mut guard = match db.0.lock() {
+                Ok(guard) => guard,
+                Err(poisoned) => poisoned.into_inner(),
+            };
+            let new_data_mutex = load_dune_data_into_memory(dune_download_file.clone())
+                .expect("could not load data into memory");
```

Comment:
If this is a long process, it might make sense to first load the data and then acquire the lock.

Also, in general its dangerous to panic (the `.expect()` here could make it happen) as it will poison the mutex making the data not accessible from other places. Additionally, it will cause the maintenance loop to stop running which may cause the service to serve stale data.